src/Features/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.Features.csproj

```diff
@@ -26,6 +26,7 @@
     <InternalsVisibleTo Include="Microsoft.VisualStudio.LanguageServices.CSharp" />
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.CSharp.EditorFeatures.Wpf" />
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.Remote.Workspaces" />
+    <InternalsVisibleTo Include="Microsoft.CodeAnalysis.ExternalAccess.Format" />
```

Comment:
Shouldn't this be a RestrictedInternalsVisibleTo like https://github.com/dotnet/roslyn/blob/e01dbb792dd07676ce20585f8ba7991dd3884d72/src/VisualStudio/Core/Def/Microsoft.VisualStudio.LanguageServices.csproj#L128? @sharwell ?

Reply:
Same applies to new IVTs added below.

Reply:
Ah, maybe, what is the difference?

Reply:
RestrictedIVT gives a namespace specific IVT access. So in the example I pointed, `ManagedSourceCodeAnalysis` only gets IVT access to types under `Microsoft.CodeAnalysis.ExternalAccess.LegacyCodeAnalysis.Api` namespace (for you will it automatically give access to `Microsoft.CodeAnalysis.ExternalAccess.Format.Api` if you have Partner="Format").

Then on the client side, you can add analyzer reference to https://www.nuget.org/packages/Microsoft.CodeAnalysis.BannedApiAnalyzers, and the analyzer will generate errors for any symbol access from client code outside the allowed namespace.

Reply:
> Shouldn't this be a RestrictedInternalsVisibleTo

No. The target of the IVT is a project in Roslyn.sln. Restricted IVTs target projects outside this repository.